### PR TITLE
Fix crosslink req id issue

### PR DESCRIPTION
**Pull request by kelvin-cao**
It was observed in a crosslink NT setup with two hosts having different
numbers of requester IDs that the traffic can go only in one direction.

In the setup, the switch of host A connects to CPU1, while the switch
of host B connects to CPU0. After loading the driver, the two hosts
have the following requester IDs. And the traffic only goes from B to A.

| Host A| Host B| 
| ---- | ----- |
|0:0.0 |0:0.0 |
|80:3.0 ||

When a TLP gose from host A to host B, it first goes from host A CPU to
the local crosslink partition (there are a pair of virtual partitions in
between the two switches in the crosslink NTB setup), with the original
requester ID be replaced with a proxy ID, then be routed to the peer
crosslink partition. If the new requester ID (prxoy ID of the local
crosslink partition) hit one entry in the requester ID table of the peer
crosslink partition, it will finally be routed to Host B. Otherwise, it
will be dropped.

The reason for the above phenomenon is that the driver wrongly registers
the requester IDs to the local crosslink partition, instead of to its
peer. So when the traffic from host A with requester ID reaches the
crosslink partition of host B and hits no entry, it goes nowhere.

This PR fix this by registering the proper requester IDs to the
peer crosslink partition. Another internal LUT window is allocated for
routing the configuration traffic to the GAS region of the peer
crosslink partition.

### CODE REVIEW

File: ntb_hw_switchtec.c [hunk 1]
```diff
@@ -1320,34 +1331,61 @@ static int switchtec_ntb_init_crosslink(struct switchtec_ntb *sndev)
...
+	if (!sndev->mmio_xlink_dbmsg_win) {
+		rc = -ENOMEM;
 		return rc;
+	}
 
-	rc = crosslink_setup_req_ids(sndev, sndev->mmio_peer_ctrl);
+	sndev->mmio_peer_dbmsg = sndev->mmio_xlink_dbmsg_win + offset;
+	sndev->nr_rsvd_luts++;
+
+
```
Comment (lsgunth) [inline ntb_hw_switchtec.c:1345]: Nit: extra blank line

**Review by lsgunth — APPROVED** [review]
Yeah, I think this makes sense to me. Slightly hard to follow the patch. Might be nice to split it up a bit more, though I'm not sure that's possible.

File: ntb_hw_switchtec.c (hunk 1 above)
Comment (kelvin-cao) [reply ntb_hw_switchtec.c:1345]: Yeah, it's a bit. I looked again, it seems to me all the changes are related to the introduction of the new LUT window, splitting them might break the consistency. I'd like to keep it as is. 

Thanks.

**Review by xyan264 — APPROVED** [review]
